x/stake/keeper/validator.go

```diff
@@ -693,6 +693,37 @@ func (k Keeper) RemoveValidator(ctx sdk.Context, address sdk.ValAddress) {
 	tstore.Set(GetTendermintUpdatesTKey(address), bz)
 }
 
+// UpdateValidatorCommission attempts to update a validator's commission rate.
+// An error is returned if the new commission rate is invalid.
+func (k Keeper) UpdateValidatorCommission(ctx sdk.Context, validator types.Validator, newRate sdk.Dec) sdk.Error {
+	commission := validator.Commission
+	blockTime := ctx.BlockHeader().Time
+
+	switch {
+	case blockTime.Sub(validator.CommissionUpdateTime).Hours() < 24:
+		// new rate cannot be changed more than once within 24 hours
+		return types.ErrCommissionUpdateTime(k.Codespace())
+
+	case newRate.LT(sdk.ZeroDec()):
+		// new rate cannot be negative
+		return types.ErrCommissionNegative(k.Codespace())
+
+	case newRate.GT(commission.MaxRate):
+		// new rate cannot be greater than the max rate
+		return types.ErrCommissionGTMaxRate(k.Codespace())
+
+	case newRate.Sub(commission.Rate).Abs().GT(commission.MaxChangeRate):
+		// new rate % change cannot be greater than the max change rate
```

Comment:
This comment is actually a bit off, it should say `new rate % points change`, we're not doing percentage change here.

We should give an example in the docs or in the comments somewhere that we are talking about capping the `percentage points` change, not the percentage change. (e.g. 1% to 2% is a 100% rate increase, but only 1 percentage point)

At first I thought we were capping percentage change, not percentage points, hence the comment.